### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by Gundersanne**
- [x] blocked on #41 
- [x] needs adjustments in frontend (`composes/$id` rather than `compose/$id`)

---

This contains mostly little adjustments to be compliant with clouddot guidelines.

They also prefer using the content-Type `application/vnd.api+json` however echo doesn't really support this, and it would require overwriting headers, or encoding the response into json ourselves etc... so I'd just leave it unless it turns out to be a real problem.

### CODE REVIEW

**Review by teg — DISMISSED** [review]
Makes sense to me. A reference to the guidelines would be nice (if it is public).

File: internal/server/api.yaml [hunk 1]
```diff
@@ -97,9 +97,31 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ComposeResponse'
-
+        '400':
+          description: the compose request is malformed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HTTPErrorList'
```
Comment (larskarlitski) [inline internal/server/api.yaml:105]: Why is this a list of errors? Seems like you only ever put one into it, no?

Comment (Gundersanne) [reply internal/server/api.yaml:105]: Yup, but it has to comply with https://jsonapi.org/format/#errors:

> Error objects MUST be returned as an array keyed by errors in the top level of a JSON:API document.

Comment (Gundersanne) [reply internal/server/api.yaml:105]: Maybe I should rename the HTTPErrorList to just HTTPErrors? because it's an object container 'errors: []'

Comment (larskarlitski) [reply internal/server/api.yaml:105]: Ah, interesting. Thanks for the link.

Both `ErrorList` and `Errors` work for me. This name is not exposed as API.

**Review by larskarlitski — APPROVED** [review]

**Pull request merged**